File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int min = Integer.MIN_VALUE;
        int z = 0;
        for(int i : nums){
            z+=i;
            min = Math.max(min,z);
            z = Math.max(z,0);
        }
        return min;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int maxSub = nums[0];
        int sum = 0;
        for(int i = 0; i < nums.Length; i++){
            if(sum < 0 )
                sum = 0;
            sum += nums[i];
            maxSub = Math.Max(maxSub, sum);
        }
        
        return maxSub;
    }
}